app/javascript/lib/validation/expression-matrices-validation.js

```diff
@@ -3,32 +3,37 @@
 */
 
 import {
-  parseLine, validateUniqueCellNamesWithinFile, ParseException
+  parseLine, validateUniqueCellNamesWithinFile, ParseException, timeOutCSFV
 } from './shared-validation'
 
 
 const whitespaceDelimiter = /\s+/
 
 /** Parse a dense matrix file */
 export async function parseDenseMatrixFile(chunker, mimeType, fileOptions) {
-  const { header, delimiter, firstTwoContentLines } = await getParsedDenseMatrixHeaderLine(chunker, mimeType)
+  const { header, delimiter, firstTwoContentLines } = await getParsedDenseMatrixHeaderLine(chunker)
 
   let issues = validateDenseHeader(header, firstTwoContentLines)
 
   // validating the header required extra lines from the file,
   // return the file reader to the first non-header line to continue validating file
   chunker.resetToFileStart()
-  await chunker.iterateLines(() => {}, 1)
+  await chunker.iterateLines({ func: () => {}, maxLines: 1 })
 
   const secondLineOfFile = firstTwoContentLines[0]
 
   const dataObj = {} // object to track multi-line validation concerns
-  await chunker.iterateLines((rawLine, lineNum, isLastLine) => {
-    const line = parseLine(rawLine, delimiter)
-    issues = issues.concat(validateValuesAreNumeric(line, isLastLine, lineNum, dataObj))
-    issues = issues.concat(validateColumnNumber(line, isLastLine, secondLineOfFile, lineNum, dataObj))
-    issues = issues.concat(validateUniqueCellNamesWithinFile(line, isLastLine, dataObj))
+  await chunker.iterateLines({
+    func: (rawLine, lineNum, isLastLine) => {
+    // check that it's still good time
+      issues = issues.concat(timeOutCSFV(chunker))
+
```

Comment:
Blocker: 100x more efficient, with negligible downside for UX:

```suggestion
  
      if (lineNum % 100) {
        // check that it's still good time
        issues = issues.concat(timeOutCSFV(chunker))
      }
```